### PR TITLE
Add tokenscript tutorial

### PR DESCRIPTION
**Pull request by colourful-land**
This part seems to be necessary for James Brown's workshop this evening, so I make a PR now. Once it's approved, please do not delete the branch since Sangalli is still working on completing it.

@hboon @James-Sangalli James Brown can't validate the knowledge in the 4th chapter because he uses Windows (can't handle Makefile). Can you run it through on your Mac to see everything works? I used `brew` command in the tutorial assuming most use OS X, but I only tested it working on my Ubuntu.

Solves #104

### CODE REVIEW

**Comment by hboon** [comment]
@colourful-land When `xmlsectool` is not installed, it's not reported (maybe the same behavior for `xmlstarlet` if it's not installed). After installing `xmlsectool`, running `make` produces these, but the signature isn't included:

```
bash-3.2$ pwd
/Users/hboon/p/alphawallet/TokenScript/examples/ticket
bash-3.2$ make
# XML Canonicalization
xmlstarlet c14n AdmissionTicket.xml  > AdmissionTicket.canonicalized.xml
# XML Validation
xmlstarlet val --xsd ../../schema/tokenscript.xsd AdmissionTicket.canonicalized.xml || (xmllint --noout --schema ../../schema/tokenscript.xsd AdmissionTicket.canonicalized.xml; rm AdmissionTicket.canonicalized.xml)
AdmissionTicket.canonicalized.xml - valid
bash-3.2$ head AdmissionTicket.canonicalized.xml
<ts:token xmlns="http://www.w3.org/1999/xhtml" xmlns:ts="http://tokenscript.org/2019/04/tokenscript">

  <ts:name xml:lang="en">Admission Ticket Token</ts:name>
  <ts:name xml:lang="zh">入場券</ts:name>
  <ts:name xml:lang="es">Boleto de admisión</ts:name>
  <ts:name form="collective" xml:lang="en">Admission Ticket Tokens</ts:name>
  <ts:name form="collective" xml:lang="zh">入場券</ts:name>
  <ts:name form="collective" xml:lang="es">Boletos de admisión</ts:name>
  <!-- TODO: will make separate Tokenscript for ERC875 -->
  <ts:contract id="erc875" interface="erc875">
```

---

```
$ brew info xmlsectool
xmlsectool: stable 2.0.0
```

Manually running this after the Makefile generates the `.out` file with the signature though:

```
xmlsectool --sign --keyInfoKeyName 'Shong Wang' --digest SHA-256 --signatureAlgorithm http://www.w3.org/2001/04/xmldsig-more#rsa-sha256 --inFile AdmissionTicket.canonicalized.xml --outFile AdmissionTicket.canonicalized.xml.out --key ssl/shong.wang.key --certificate ssl/shong.wang.certs
```

**Comment by colourful-land** [comment]
Okay, try it again. I regret to change the way make works and updated the docs for the new process (now you need to type the filename) but that's needed to make the Makefile portable (since it now can be used with user-created tokenscript without modification).

Unless you have a better method of course.

**Comment by colourful-land** [comment]
@JamesSmartCell I changed the reviewer to you because I want you to correct the command here in the last bit "I don't use make":

https://github.com/AlphaWallet/TokenScript/blob/add-tokenscript-tutorial/tutorial/Step%204:%20Signing%20and%20publishing.md

Once you put the correct command on Windows, replacing the old command, do make sure the indent is correct. Mine wasn't done correctly but I can't be bothered to fix it because you are to put the right comamnd in anyway.

**Comment by colourful-land** [comment]
@James-Sangalli you mentioned the content needs to be integrated into the rest of your writing. Can you do it? You can edit the content I wrote freely.

**Comment by hboon** [comment]
> Okay, try it again @hboon I regret to change the way make works and updated the docs for the new process (now you need to type the filename) but that's needed to make the Makefile portable (since it now can be used with user-created tokenscript without modification).
> 
> Unless you have a better method of course.

Works great for me now. You could have `make canonicalized <filename.xml>` and `make sign <filename.canonicalized.xml>` and make the latter dependent on the former but maybe that makes the script too complicated?

**Comment by colourful-land** [comment]
@hboon typically make work in the way that you specify the target file (the one that doesn't exist), not the source file. I don't know how to make your syntax work like this.

    $ make canonicalized <filename.xml>

**Comment by hboon** [comment]
@colourful-land `make` seems to support this:

```
make input=AdmissionTicket.xml canonicalize
```

and:

```
make input=AdmissionTicket.canoicalized.xml sign
```

The order looks a bit strange, but my `make`-fu is weak and maybe that is normal? The main usability problem seems to be having to make the Makefile portable for an unknown input file.

If you think the above (`make input=AdmissionTicket.xml canonicalize`) is good enough and can be done quickly, please go ahead, otherwise let's leave it for now and see if there's any feedback or if we have a better idea?

**Comment by colourful-land** [comment]
@hboon Typically variables are used for libraries and not for source files so I don't know of a rule that can create value `AdmissionTicket.canonicalized.xml`, so instead the output would be `AdmissionTicket.xml.canonicalized` since concatenation is always easier, which is a bit weird.

I'll switch the focus to a video tutorial based or GUI based method for signing. The text tutorial was quickly made to be the first line of defence (think macrophage before the while blood cell comes)

File: tutorial/Step 4: Debugging and testing.md [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
+After cloning the source code you can run it with either the emulator or your own phone via xcode or AppCode. If you are using the emulator, simply copy the path it prints when loading up in xcode and cd into it. Once there you shoud see a directory called 'assetDefinitions', simply drop the file into this directory.
 
 If you are using your real device, simply airdrop the .tsml or .shtml, .css and .xml files. 
-** Make sure you name the file with the contract address followed by the extension **. 
 
 ### Debugging on Android
-Like with iOS, you can either use your own device or the emulator.
-To use the emulator, simply run the GitHub project via Android Studio and run it. 
-You can drag and drop the file in the app_data/AlphaWallet directory. 
+Like with iOS, you can either use your own device or the emulator. To use the emulator, simply run the GitHub project via Android Studio and run it. You can drag and drop the file in the app_data/AlphaWallet directory.
```
Comment (James-Sangalli) [inline tutorial/Step 4: Debugging and testing.md]: @JamesSmartCell Please check this to make sure it is accurate, if not, please make the quick change.

**Comment by colourful-land** [comment]
@James-Sangalli Why is Signing before testing on Android / iOS? You should test before signing shouldn't you?

File: tutorial/Step 4: Debugging and testing.md (hunk 1 above)
Comment (JamesSmartCell) [reply tutorial/Step 4: Debugging and testing.md]: I updated the text please check.

**Pull request merged**